internal/xds/translator/translator.go

```diff
@@ -632,10 +683,20 @@ func (t *Translator) addRouteToRouteConfig(
 		}
 	}
 	xdsRouteCfg.VirtualHosts = append(xdsRouteCfg.VirtualHosts, vHostList...)
-
+	// TODO(zhaohuabing) we need to sort the virtual hosts from the most specific to the least specific
```

Comment:
we need to sort the virtual hosts from the most specific to the least specific to ensure that the most specific virtual host is matched first.

Reply:
Since this PR is already very large, created a seperate PR to fix this: https://github.com/envoyproxy/gateway/pull/6603